test/printf.jl

```diff
@@ -1,45 +1,121 @@
 # This file is a part of Julia. License is MIT: http://julialang.org/license
 
-# printf
-# int
 @test (@sprintf "%d" typemax(Int64)) == "9223372036854775807"
-for (fmt, val) in (("%i", "42"),
-                   ("%u", "42"),
-                   ("Test: %i", "Test: 42"),
-                   ("%#x", "0x2a"),
-                   ("%#o", "052"),
-                   ("%X", "2A"),
-                   ("%X", "2A"),
-                   ("% i", " 42"),
-                   ("%+i", "+42"),
-                   ("%4i", "  42"),
-                   ("%-4i", "42  "))
-    @test( @eval(@sprintf($fmt, 42) == $val))
+
+# printing an int value
+for (fmt, val) in (
+        ("%i", "42"),
+        ("%i", "42"),
+        ("%u", "42"),
+        ("%x", "2a"),
+        ("%X", "2A"),
+        ("%o", "52"),
+        ("%a", "0x1.5p+5"),
+        ("%A", "0X1.5P+5"),
+        ("%f", "42.000000"),
+        ("%g", "42"),
+        ("%p", "0x000000000000002a"),
+    ),
+      num in (UInt16(42), UInt32(42), UInt64(42), UInt128(42),
+              Int16(42), Int32(42), Int64(42), Int128(42),
+              #big"42" # causes stack overflow on %a ; gh #14409
+    )
+    @test( @eval(@sprintf($fmt, $num) == $val))
```

Comment:
The convention part is just the lack of parens with spaces inside of them. I don't think we have a convention around whether `@test` or `@eval` comes first but I agree that having the `@eval` outside seems (marginally) better.